include/zephyr/spinlock.h

```diff
@@ -29,6 +29,56 @@ extern "C" {
  * @{
  */
 
+/**
```

Comment:
This change must be placed in its own PR.

Reply:
See PR description - this change already is in a separate change set. That's why the PR is a draft. Depending on the outcome of the other PR this PR will either stay as it is or be changed to use the classical spinlock pattern.